src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -4,274 +4,213 @@
+                lines = await connection.ReadRequestHeaderAndSendResponseAsync(response);
+            });
 
             return lines;
         }
 
-        public static async Task<bool> WebSocketHandshakeAsync(Socket s, StreamReader reader, StreamWriter writer)
+        public Task<List<string>> AcceptConnectionSendDefaultResponseAndCloseAsync()
         {
-            string serverResponse = null;
-            string currentRequestLine;
-            while (!string.IsNullOrEmpty(currentRequestLine = await reader.ReadLineAsync().ConfigureAwait(false)))
-            {
-                string[] tokens = currentRequestLine.Split(new char[] { ':' }, 2);
-                if (tokens.Length == 2)
-                {
-                    string headerName = tokens[0];
-                    if (headerName == "Sec-WebSocket-Key")
-                    {
-                        string headerValue = tokens[1].Trim();
-                        string responseSecurityAcceptValue = ComputeWebSocketHandshakeSecurityAcceptValue(headerValue);
-                        serverResponse =
-                            "HTTP/1.1 101 Switching Protocols\r\n" +
-                            "Upgrade: websocket\r\n" +
-                            "Connection: Upgrade\r\n" +
-                            "Sec-WebSocket-Accept: " + responseSecurityAcceptValue + "\r\n\r\n";
-                    }
-                }
-            }
-
-            if (serverResponse != null)
-            {
-                // We received a valid WebSocket opening handshake. Send the appropriate response.
-                await writer.WriteAsync(serverResponse).ConfigureAwait(false);
-                return true;
-            }
-
-            return false;
+            return AcceptConnectionSendResponseAndCloseAsync(GetDefaultHttpResponse());
         }
 
-        private static string ComputeWebSocketHandshakeSecurityAcceptValue(string secWebSocketKey)
+        public async Task AcceptConnectionAsync(Func<Connection, Task> funcAsync)
         {
-            // GUID specified by RFC 6455.
-            const string Rfc6455Guid = "258EAFA5-E914-47DA-95CA-C5AB0DC85B11";
-            string combinedKey = secWebSocketKey + Rfc6455Guid;
-
-            // Use of SHA1 hash is required by RFC 6455.
-            SHA1 sha1Provider = new SHA1CryptoServiceProvider();
-            byte[] sha1Hash = sha1Provider.ComputeHash(Encoding.UTF8.GetBytes(combinedKey));
-            return Convert.ToBase64String(sha1Hash);
-        }
+            Socket s = await _listenSocket.AcceptAsync().ConfigureAwait(false);
 
-        public static async Task<List<string>> AcceptSocketAsync(Socket server, Func<Socket, Stream, StreamReader, StreamWriter, Task<List<string>>> funcAsync, Options options = null)
-        {
-            options = options ?? new Options();
-            Socket s = await server.AcceptAsync().ConfigureAwait(false);
-            s.NoDelay = true;
-            try
+            using (s)
```

Comment:
Nit: just curious why this isn't just:
```C#
using (Socket s = await ...)
```
?

Reply:
No good reason, I'll fix